lightning/src/routing/router.rs

```diff
@@ -1086,21 +1140,24 @@ impl<'a> CandidateRouteHop<'a> {
 
 	fn effective_capacity(&self) -> EffectiveCapacity {
 		match self {
-			CandidateRouteHop::FirstHop { details } => EffectiveCapacity::ExactLiquidity {
+			CandidateRouteHop::FirstHop { details, .. } => EffectiveCapacity::ExactLiquidity {
 				liquidity_msat: details.next_outbound_htlc_limit_msat,
 			},
 			CandidateRouteHop::PublicHop { info, .. } => info.effective_capacity(),
-			CandidateRouteHop::PrivateHop { hint: RouteHintHop { htlc_maximum_msat: Some(max), .. }} =>
+			CandidateRouteHop::PrivateHop { hint: RouteHintHop { htlc_maximum_msat: Some(max), .. }, ..} =>
 				EffectiveCapacity::HintMaxHTLC { amount_msat: *max },
-			CandidateRouteHop::PrivateHop { hint: RouteHintHop { htlc_maximum_msat: None, .. }} =>
+			CandidateRouteHop::PrivateHop { hint: RouteHintHop { htlc_maximum_msat: None, .. }, ..} =>
 				EffectiveCapacity::Infinite,
 			CandidateRouteHop::Blinded { hint, .. } =>
 				EffectiveCapacity::HintMaxHTLC { amount_msat: hint.0.htlc_maximum_msat },
 			CandidateRouteHop::OneHopBlinded { .. } => EffectiveCapacity::Infinite,
 		}
 	}
 
-	fn id(&self, channel_direction: bool /* src_node_id < target_node_id */) -> CandidateHopId {
+	///  Returns the id of this hop.
+	///  For `Blinded` and `OneHopBlinded` we return `CandidateHopId::Blinded` because we don't know the channel id.
+	///  For any other option we return `CandidateHopId::Clear` because we know the channel id and the direction.
+	pub fn id(&self, channel_direction: bool /* src_node_id < target_node_id */) -> CandidateHopId {
```

Comment:
Does this need to be `pub`? If so, we should document `channel_direction` in the method docs. I'd rather not expose this detail if we don't need to, though.

Reply:
If I am not mistaken, @TheBlueMatt wanted this to be `pub`?

Reply:
Ah, is that referring to https://github.com/lightningdevkit/rust-lightning/pull/2551#discussion_r1316176756 ? I'd at least like to rid ourselves of the direction parameter if we want to expose it for some reason.